src/sfe_ublox_cellular.cpp

```diff
@@ -4682,76 +4682,120 @@ UBX_CELL_error_t SparkFun_ublox_Cellular::getFileContents(String filename, char
     return err;
 }
 
-UBX_CELL_error_t SparkFun_ublox_Cellular::getFileBlock(const String &filename, char *buffer, size_t offset, size_t requested_length,
-                                        size_t &bytes_read)
-{
-    bytes_read = 0;
-    if (filename.length() < 1 || buffer == nullptr || requested_length < 1)
+UBX_CELL_error_t SparkFun_ublox_Cellular::getFileBlock(const String &filename, char *buffer, size_t offset, size_t requestedLength,
+                                        size_t &bytesRead)
+{
+  UBX_CELL_error_t err;
+  char *command;
+  char *response;
+
+  bytesRead = 0;
+  if (filename.length() < 1 || buffer == nullptr || requestedLength < 1)
+  {
+      return UBX_CELL_ERROR_UNEXPECTED_PARAM;
+  }
+
+  // trying to get a byte at a time does not seem to be reliable so this method must use
+  // a real UART.
+  if (_hardSerial == nullptr)
+  {
+    if (_printDebug == true)
     {
-        return UBX_CELL_ERROR_UNEXPECTED_PARAM;
+      _debugPort->println(F("getFileBlock: only works with a hardware UART"));
     }
+    return UBX_CELL_ERROR_INVALID;
+  }
+
+  command = ubx_cell_calloc_char(strlen(UBX_CELL_FILE_SYSTEM_READ_BLOCK) + filename.length() + 28);
+  if (command == nullptr)
+  {
+    return UBX_CELL_ERROR_OUT_OF_MEMORY;
+  }
+  sprintf(command, "%s=\"%s\",%lu,%lu", UBX_CELL_FILE_SYSTEM_READ_BLOCK, filename.c_str(), (unsigned long) offset, (unsigned long) requestedLength);
 
-    // trying to get a byte at a time does not seem to be reliable so this method must use
-    // a real UART.
-    if (_hardSerial == nullptr)
+  response = ubx_cell_calloc_char(minimumResponseAllocation);
+  if (response == nullptr)
+  {
+    if (_printDebug == true)
     {
-        if (_printDebug == true)
-        {
-            _debugPort->println(F("getFileBlock: only works with a hardware UART"));
-        }
-        return UBX_CELL_ERROR_INVALID;
+      _debugPort->print(F("getFileBlock: response alloc failed: "));
+      _debugPort->println(minimumResponseAllocation);
     }
-
-    size_t cmdLen = filename.length() + 32;
-    char *cmd = ubx_cell_calloc_char(cmdLen);
-    if (cmd == nullptr)
-        return UBX_CELL_ERROR_OUT_OF_MEMORY;
-    snprintf(cmd, cmdLen, "at+urdblock=\"%s\",%zu,%zu\r\n", filename.c_str(), offset, requested_length);
-    sendCommand(cmd, false);
-
-    int ich;
-    char ch;
-    int quote_count = 0;
-    size_t comma_idx = 0;
-
-    while (quote_count < 3)
+    free(command);
+    return UBX_CELL_ERROR_OUT_OF_MEMORY;
+  }
```

Comment:
`response` is a fixed size, so don't use `ubx_cell_calloc_char()`, just create it as an array:

`char response[minimumResponseAllocation];`